File: Java/167. Two Sum II - Input Array Is Sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int a = numbers.length - 1, b = 0;
        while(numbers[a] + numbers[b] != target){
            if(numbers[a] + numbers[b] > target){ a--; }
            else{ b++; }
        }
        int[] result = {b + 1, a + 1};
        return result;
    }
}

File: Kotlin/167. Two Sum II - Input Array Is Sorted.kt
class Solution {
    fun twoSum(numbers: IntArray, target: Int): IntArray {
        var a = numbers.size - 1
        var b = 0
        while(numbers[a] + numbers[b] != target){
            if(numbers[a] + numbers[b] > target){ a-- }
            else{ b++ }
        }
        return intArrayOf(b + 1, a + 1)
    }
}
